src/app/clusters/valve-configuration-and-control-server/valve-configuration-and-control-cluster.cpp

```diff
@@ -18,521 +18,500 @@
+    }
 
-    if (CHIP_NO_ERROR != error)
+    // If there was an error, we know nothing about the current state
+    if (err != CHIP_NO_ERROR)
     {
-        ChipLogError(Zcl, "Unable to emit ValveFault event [ep=%d]", ep);
-        return error;
+        SaveAndReportIfChanged(mCurrentLevel, DataModel::NullNullable, Attributes::CurrentLevel::Id);
+        SaveAndReportIfChanged(mCurrentState, DataModel::NullNullable, Attributes::CurrentState::Id);
     }
 
-    ChipLogProgress(Zcl, "Emit ValveFault event [ep=%d]", ep);
-    return CHIP_NO_ERROR;
+    // Emit the Transition state.
+    emitValveChangeEvent(ValveStateEnum::kTransitioning);
+
+    return err;
 }
 
-static void onValveConfigurationAndControlTick(System::Layer * systemLayer, void * data)
+std::optional<DataModel::ActionReturnStatus>
+ValveConfigurationAndControlCluster::HandleOpenCommand(const DataModel::InvokeRequest & request, TLV::TLVReader & input_arguments,
+                                                       CommandHandler * handler)
 {
-    RemainingDurationTable * item = reinterpret_cast<RemainingDurationTable *>(data);
-    VerifyOrReturn(item != nullptr, ChipLogError(Zcl, "Error retrieving RemainingDuration item"));
-
-    DataModel::Nullable<uint32_t> rDuration = item->remainingDuration;
-    VerifyOrReturn(!rDuration.IsNull());
+    Commands::Open::DecodableType commandData;
+    DataModel::Nullable<chip::Percent> openTargetLevel;
+    DataModel::Nullable<uint32_t> openDuration;
 
-    EndpointId ep = item->endpoint;
+    ReturnErrorOnFailure(commandData.Decode(input_arguments));
 
-    if (rDuration.Value() > 0)
+    // Check if there is any Fault registered
+    if (mValveFault.HasAny())
     {
-        SetRemainingDuration(ep, DataModel::MakeNullable<uint32_t>(--rDuration.Value()));
-        startRemainingDurationTick(ep);
+        ReturnErrorOnFailure(handler->AddClusterSpecificFailure(request.path, to_underlying(StatusCodeEnum::kFailureDueToFault)));
+        return std::nullopt;
     }
-    else
+
+    // Check if the provided openDuration has a value
+    // Save the duration if provided, it can be Null or an actual value.
+    // If no value is provided, use the DefaultOpenDuration attribute value.
+    openDuration = commandData.openDuration.HasValue() ? commandData.openDuration.Value() : mDefaultOpenDuration;
+
+    // Check rules for TargetLevel if enabled
+    if (mFeatures.Has(Feature::kLevel))
     {
-        SetRemainingDurationNull(ep);
+        ReturnErrorOnFailure(GetAdjustedTargetLevel(commandData.targetLevel, openTargetLevel));
     }
-}
 
-void startRemainingDurationTick(EndpointId ep)
-{
-    RemainingDurationTable * item = GetRemainingDurationItem(ep);
-    VerifyOrReturn(item != nullptr, ChipLogError(Zcl, "Error retrieving RemainingDuration item"));
+    // Use the SetValveLevel function to handle the setting of internal values.
+    ReturnErrorOnFailure(SetValveLevel(openTargetLevel, openDuration));
```

Comment:
```suggestion
    return SetValveLevel(openTargetLevel, openDuration);
```

Should it just be that since otherwise we return success?